packages/api/src/routers/auth/auth_router.ts

```diff
@@ -373,18 +374,27 @@ export function authRouter() {
     '/email-login',
     cors<express.Request>(corsConfig),
     async (req: express.Request, res: express.Response) => {
-      const { email, password } = req.body
-
-      if (!email || !password) {
+      interface LoginRequest {
+        email: string
+        password: string
+      }
+      function isValidLoginRequest(obj: any): obj is LoginRequest {
+        return (
+          'email' in obj &&
+          obj.email.trim().length > 0 && // email must not be empty
+          'password' in obj &&
+          obj.password.length >= 8 // password must be at least 8 characters
+        )
+      }
+      if (!isValidLoginRequest(req.body)) {
         return res.redirect(
           `${env.client.url}/auth/email-login?errorCodes=${LoginErrorCode.InvalidCredentials}`
         )
       }
-
+      const { email, password } = req.body
       try {
-        const models = initModels(kx, false)
-        const user = await models.user.getWhere({
-          email,
+        const user = await getRepository(User).findOneBy({
+          email: ILike(email.trim()), // case insensitive
```

Comment:
Do we need to use ILike here? I worry that maybe it could do partial matches?

Reply:
Yeah, partial match could be a problem.

I'm trying to do a case insensitive match of the email address. Probably having`lower(col) = lower(value)` in the query is better

@jacksonh

Reply:
Oh yes, i think using lower is kind of the standard for this.